### DISCUSSION TITLE
subclassing pieces to change move quality

### DISCUSSION THREAD

**Discussion by NathanTech7713** [niklasf/python-chess#823] [Q&A]
Hi there,

I wondered if someone would be able to talk me through the best way to approach modifying pieces so that they have extra moves?
An example is in the variation Sideways Pawns where pawns can move, well, sideways.
Thanks in advance.

**Comment by niklasf** [comment]
Hi, unfortunately the rules for piece movement in this library are implemented pretty rigidly. So there is no good way to easily override them or add new piece types.
